controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -422,15 +437,19 @@ export class FederatedGraphRepository {
   }
 
   public async isLatestValidRouterConfigVersion(targetId: string, schemaVersionId: string): Promise<boolean> {
-    const latestVersion = await this.db.query.schemaVersion.findFirst({
-      columns: {
-        id: true,
-      },
-      where: and(eq(schema.schemaVersion.targetId, targetId), eq(schema.schemaVersion.isComposable, true)),
-      orderBy: desc(schema.schemaVersion.createdAt),
-    });
+    const latestValidVersion = await this.db
+      .select({
+        id: schemaVersion.id,
+        routerConfig: graphCompositions.routerConfig,
```

Comment:
This method should not fetch the router config. It is a lightweight check.

Reply:
Done